packages/suite/src/components/suite/modals/AddToken/index.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import TrezorConnect, { TokenInfo } from 'trezor-connect';
+import { Input, Button } from '@trezor/components';
+import * as tokenActions from '@wallet-actions/tokenActions';
+import { Modal, QuestionTooltip } from '@suite-components';
+import { Translation } from '@suite-components/Translation';
+import { useActions, useSelector, useTranslation } from '@suite-hooks';
+import { isAddressValid } from '@wallet-utils/validation';
+
+const Wrapper = styled.div`
+    display: flex;
+    justify-content: center;
+    margin: 24px 0px;
+`;
+const Actions = styled.div`
+    display: flex;
+    justify-content: center;
+`;
+interface Props {
+    onCancel: () => void;
+}
+
+const AddToken = (props: Props) => {
+    const [contractAddress, setContractAddress] = useState<string>('');
+    const [tokenInfo, setTokenInfo] = useState<TokenInfo[] | undefined>(undefined);
+    const [isFetching, setIsFetching] = useState(false);
+    const { translationString } = useTranslation();
+    const [error, setError] = useState<string | null>(null);
+    const { addToken } = useActions({
+        addToken: tokenActions.addToken,
+    });
+    const selectedAccount = useSelector(state => state.wallet.selectedAccount);
+    const { account } = selectedAccount;
+
+    useEffect(() => {
+        const loadTokenInfo = async () => {
```

Comment:
I think this can be defined outside of `useEffect` while using arguments `account, contractAddress, error, translationString`

Reply:
hmm I am not sure if it worth the trouble passing all arguments to the function in this case. But sure, I'll update it

<img width="816" alt="Screenshot 2020-12-17 at 17 41 38" src="https://user-images.githubusercontent.com/6961901/102516432-1f9fcf00-408f-11eb-8090-10d22cb9dd91.png">